src/udev/udev-event.c

```diff
@@ -346,12 +346,30 @@ size_t udev_event_apply_format(struct udev_event *event,
                         } else if (from[0] == '%') {
                                 /* substitute format char */
                                 unsigned int i;
+                                unsigned int num = 0;
+                                char from1 = from[1];
 
                                 if (from[1] == '%') {
                                         from++;
                                         goto copy;
                                 }
 
+                                /* extract possible format length and move from behind it*/
+                                if (isdigit(from1)) {
+                                        char *tail;
+
+                                        from++;
+                                        num = (int) strtoul(from, &tail, 10);
+                                        if (num > 0) {
+                                                log_debug("format length='%i'", num);
+                                                from = strjoina("%", tail);
+                                                l = num+1;
```

Comment:
It seems that the length of `src` might be greater than the length of `dst` in this case.